src/flag_gems/ops/fft_1d.py

```diff
@@ -0,0 +1,209 @@
+import torch
+import torch.profiler
+import triton
+import triton.language as tl
+
+
+@triton.jit
+def bit_reverse_kernel(real_in, imag_in, real_out, imag_out, n):
+    """先进行逆序排列: input[i] -> output[bit_reverse(i)]"""
+    tid = tl.program_id(0)
+    if tid >= n:
+        return
+
+    # 计算位数
+    temp_n = n
+    bit_count = 0
+    while temp_n > 1:
+        temp_n //= 2
+        bit_count += 1
+
+    # 计算逆序
+    idx = tid
+    rev_idx = 0
+    temp_idx = idx
+    for i in range(bit_count):
+        rev_idx = (rev_idx << 1) | (temp_idx & 1)
+        temp_idx = temp_idx >> 1
+
+    # input[idx] -> output[rev_idx]
+    val_real = tl.load(real_in + idx)
+    val_imag = tl.load(imag_in + idx)
+    tl.store(real_out + rev_idx, val_real)
+    tl.store(imag_out + rev_idx, val_imag)
+
+
+@triton.jit
+def fft_stage_kernel(real_ptr, imag_ptr, n, stage):
+    """迭代 FFT 阶段"""
+    pi = 3.141592653589793238462643383279502884  # 圆周率
+    tid = tl.program_id(0)
+
+    if tid >= n // 2:
+        return
+
+    # 计算当前阶段的参数
+    block_size = 1 << stage  # 2^stage
+    half_block = block_size // 2
+
+    # 每个线程处理一个蝶形对
+    butterfly_group = tid // half_block  # 属于第几个大块
+    pos_in_group = tid % half_block  # 在块前半部分的位置
+
+    # 计算蝶形对中两个元素的索引
+    first_idx = butterfly_group * block_size + pos_in_group
+    second_idx = first_idx + half_block
+
+    if second_idx >= n:
+        return
+
+    # load
+    a_real = tl.load(real_ptr + first_idx)
+    a_imag = tl.load(imag_ptr + first_idx)
+    b_real = tl.load(real_ptr + second_idx)
+    b_imag = tl.load(imag_ptr + second_idx)
+
+    # 计算复数幅度
+    angle = -2.0 * pi * pos_in_group / block_size
+    w_real = tl.cos(angle)
+    w_imag = tl.sin(angle)
+
+    # 使用幅度
+    tw_real = b_real * w_real - b_imag * w_imag
+    tw_imag = b_real * w_imag + b_imag * w_real
+
+    # 蝶形运算
+    result_a_real = a_real + tw_real
+    result_a_imag = a_imag + tw_imag
+    result_b_real = a_real - tw_real
+    result_b_imag = a_imag - tw_imag
+
+    # store
+    tl.store(real_ptr + first_idx, result_a_real)
+    tl.store(imag_ptr + first_idx, result_a_imag)
+    tl.store(real_ptr + second_idx, result_b_real)
+    tl.store(imag_ptr + second_idx, result_b_imag)
+
+
+def fft_1d(x: torch.Tensor, output: torch.Tensor) -> torch.Tensor:
+    N = x.shape[0]
+    assert N > 0 and (N & (N - 1)) == 0  # 确保N是2的整数幂
+
+    x_real = x.real.clone()
+    x_imag = x.imag.clone()
+
+    # 位逆序
+    temp_real = torch.zeros_like(x_real)
+    temp_imag = torch.zeros_like(x_imag)
+    bit_reverse_kernel[(N,)](x_real, x_imag, temp_real, temp_imag, N)
+
+    x_real.copy_(temp_real)
+    x_imag.copy_(temp_imag)
+
+    log2n = N.bit_length() - 1
+    for stage in range(1, log2n + 1):
+        fft_stage_kernel[(N // 2,)](x_real, x_imag, N, stage)
+
+    output.real.copy_(x_real)
+    output.imag.copy_(x_imag)
+    return output
+
+
+# def torch_fft(x):
```

Comment:
Please avoid commenting large chunk of source code in this change.

- If the code after this line is no longer needed, just remove them
- If the code after this line belongs to somethere else, please find a new home for them.
- If these code are supposed to be reused as a benchmark, please move the logic to the benchmark module. This can be left to a future PR, but filing an issue for this would be a good practice.